### ISSUE TITLE
Merge the injection token for cache and cache it self

### ISSUE THREAD

**Issue by snys98** [nestjs/nest#12960]
### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

for now we inject cacheManager in this way
```
import { CACHE_MANAGER} from '@nestjs/cache-manager';
import { Cache} from 'cache-manager';

@Controller()
export class AsyncRegisterController {
  constructor(@Inject(CACHE_MANAGER) private cacheManager: Cache) {}
```
which is hard to remember or code, because:
The `Cache` class is imported from the `cache-manager`, while `CACHE_MANAGER` token from the `@nestjs/cache-manager` package.

### Describe the solution you'd like

we can simply wrap the `Cache` from `cache-manager` with 
- cache.constants.ts
```tyescirpt 
import { Cache } from "cache-manager";

export type NestCache = Cache;
export const CACHE_MANAGER = 'CACHE_MANAGER';
export const NestCache = CACHE_MANAGER;
```
this solution is not following correct structure, just to demonstrate my idea

### Teachability, documentation, adoption, migration strategy

existing code will not be affected, all we need to do is to update documentation and mark CACHE_MANAGER as deprecated?

### What is the motivation / use case for changing the behavior?

the motivation is that we can simplify our injection code to
```typescript
import { NestCache } from '@nestjs/cache-manager';

@Controller()
export class AsyncRegisterController {
  constructor(@Inject(NestCache) private cacheManager: NestCache) {}
```
which is much easier and cleanner

**Comment by micalevisk** [comment]
I think the the "issue" with having that indirection (`NestCache` type) is that isn't clear that the cache manager is from cache-manager lib. But I'm fine with abstracting it away

Also, I'd suggest this one instead:

```ts
import { Cache } from '@nestjs/cache-manager';

@Controller()
export class Foo {
  constructor(private cacheManager: Cache) {}
```

<details>
<summary>yeah, it works</summary>

![image](https://github.com/nestjs/nest/assets/13461315/204bca7a-a967-4128-90d5-ee4afcf2c8ad)


</details>

**Comment by snys98** [comment]
> I think the the "issue" with having that indirection (`NestCache` type) is that isn't that clear that the cache manager is from `cache-manager` lib. But I'm fine with abstracting it away
> 
> Also, I'd suggest this one instead:
> 
> ```ts
> import { Cache } from '@nestjs/cache-manager';
> 
> @Controller()
> export class Foo {
>   constructor(private cacheManager: Cache) {} // no need to recall the token
> ```
> 
> yeah, it works

yes, I think either way is ok, it's just better to import 1 thing from 1 lib than 2 things from 2 libs

**Comment by jmcdo29** [comment]
_Personally_, I would rather import the specific token from the Nest wrapper and the actual type from the implementation library. Yes, it's another library to have to worry about, in terms of making sure it comes from the right place, but it also means that Nest can use [`peerDependencies`](https://github.com/nestjs/cache-manager/blob/master/package.json#L53) for the library and allow devs to install whichever version of the library it needs, meaning less maintenance on our side when updates are released.

**Comment by snys98** [comment]
> _Personally_, I would rather import the specific token from the Nest wrapper and the actual type from the implementation library. Yes, it's another library to have to worry about, in terms of making sure it comes from the right place, but it also means that Nest can use [`peerDependencies`](https://github.com/nestjs/cache-manager/blob/master/package.json#L53) for the library and allow devs to install whichever version of the library it needs, meaning less maintenance on our side when updates are released.

From my understanding, this library is tightly related to cache-manager and it's appropriate to link cache manager as a peer dependency. It will cause trouble when user installed very old or very new versions, if he didn't know what the version that @nest/cache-manager is actually using.

**Comment by jmcdo29** [comment]
[We already use peerDeps](https://github.com/nestjs/cache-manager/blob/56cda725af44ae4be85eb6ec674a9df67b45b3be/package.json#L53), so as long as `cache-manager<=5` is installed, this library will work. If we set to `dependencies` then npm/yarn/pnpm will auto-install the package
